### ISSUE TITLE
Error when using vision_deficiency

### ISSUE THREAD

**Issue by julianwegkamp** [Studiosity/grover#127]
Hi,

When setting vision_deficiency (I did because I blindlessly copied the example config in the README), processor.js will return an error which leads to Grover::JavaScript::ReferenceError: type is not defined.

I believe that line 106 it should be `page.emulateVisionDeficiency(visionDeficiency);`

https://github.com/Studiosity/grover/blob/33b05b39086320799feb39bf4ca1daa9e10df353/lib/grover/js/processor.js#L103-L107

I you like, I can make a PR with a fix.

Best regards,
Julian

**Comment by abrom** [comment]
Ahh.. yes that's a typo! I suspect more likely a copy paste out of the documentation.. A PR would be great!

Better yet would be a spec test to validate it's working as expected (i'm a little surprised I hadn't added one myself). The test would seem to be pretty straight forward. If the `vision_deficiency` setting was set to `deuteranopia` then we can just set the background to say red then see that the result of `to_png` returns an image that is an olive type colour

Something like:
```
      context 'when vision deficiency is set to `deuteranopia`' do
        let(:url_or_html) { '<html><body style="background-color: red"></body></html>' }
        let(:options) { { 'visionDeficiency' => 'deuteranopia' } }

        it { expect(convert.unpack('C*')).to start_with "\x89PNG\r\n\x1A\n".unpack('C*') }
        it { expect(image.type).to eq 'PNG' }
        it { expect(image.dimensions).to eq [800, 600] }
        it { expect(mean_colour_statistics(image)).to eq %w[109 90 28] }
      end
```
inserted somewhere around here: https://github.com/Studiosity/grover/blob/main/spec/grover/processor_spec.rb#L827

Please note I've not actually tested those colour statistics.. they are an approximation. But the real numbers should be pretty close to that!

**Comment by abrom** [comment]
Ahh.. actually it looks like the `page.emulateVisionDeficiency` method was only added in v3.2.0

https://github.com/puppeteer/puppeteer/releases/tag/v3.2.0

However I have a helper for that. If you wrap the context in:
```
      # Only test `emulateVisionDeficiency` if the Puppeteer supports it
      if puppeteer_version_on_or_after? '3.2.0'
        context ....
      end
```

**Comment by abrom** [comment]
Fixed in #128
